src/bin.rs

```diff
@@ -1,508 +1,658 @@
-                &self.collect(),
-                self.mean,
-                self.std_dev,
-                self.iterations,
-                self.luminosity,
-                SEED,
-            )
-            .generate_lut(self.level),
-            Algorithm::NearestNeighbor => {
-                NearestNeighborRemapper::new(&self.collect(), self.luminosity)
-                    .generate_lut(self.level)
-            },
-        };
+#[derive(Bpaf, Clone, Debug)]
+struct GaussianRbfArgs {
+    /// Shape parameter for the default Gaussian RBF interpolation. Effectively creates more or
+    /// less blending between colors in the palette, where bigger numbers equal less blending.
+    /// Effect is heavily dependant on the number of nearest colors used.
+    #[bpaf(short, long, argument("SHAPE"), fallback(128.0), display_fallback)]
+    shape: f64,
+    #[bpaf(external)]
+    common_rbf: CommonRbf,
+    #[bpaf(external)]
+    common: Common,
+}
 
-        sp.stop_and_persist("✔", format!("Generated {name} LUT in {:?}", time.elapsed()));
+#[derive(Bpaf, Clone, Debug)]
+enum LutAlgorithm {
+    GaussianRbf(#[bpaf(external(gaussian_rbf_args))] GaussianRbfArgs),
+    #[bpaf(adjacent)]
+    ShepardsMethod {
+        /// Enable using Shepard's method (Inverse Distance RBF) for interpolation.
+        #[bpaf(short('S'), long("shepards-method"), req_flag(()))]
+        _shepards_method: (),
+        /// Power parameter for shepard's method.
+        #[bpaf(short, long, argument("POWER"), fallback(4.0), display_fallback)]
+        power: f64,
+        #[bpaf(external)]
+        common_rbf: CommonRbf,
+        #[bpaf(external)]
+        common: Common,
+    },
+    #[bpaf(adjacent)]
+    GaussianSampling {
+        /// Enable using Gaussian sampling for interpolation (slow).
+        #[bpaf(short('G'), long("gaussian-sampling"), req_flag(()))]
+        _gaussian_sampling: (),
+        /// Average amount of noise to apply in each iteration.
+        #[bpaf(short, long, argument("MEAN"), fallback(0.0), display_fallback)]
+        mean: f64,
+        /// Standard deviation parameter for the noise applied in each iteration.
+        #[bpaf(short, long, argument("STD_DEV"), fallback(20.0), display_fallback)]
+        std_dev: f64,
+        /// Number of iterations of noise to apply to each pixel.
+        #[bpaf(short, long, argument("ITERS"), fallback(512), display_fallback)]
+        iterations: usize,
+        /// Seed for noise rng.
+        #[bpaf(
+            short('S'),
+            long,
+            argument("SEED"),
+            fallback(42080085),
+            display_fallback
+        )]
+        seed: u64,
+        #[bpaf(external)]
+        common: Common,
+    },
+    #[bpaf(adjacent)]
+    NearestNeighbor {
+        /// Disable interpolation completely.
+        #[bpaf(short('N'), long("nearest-neighbor"), req_flag(()))]
+        _nearest_neighbor: (),
+        #[bpaf(external)]
+        common: Common,
+    },
+    #[bpaf(skip)]
```

Comment:
Good catch, I conditionally allow this variant in the utility below since it's used in one command but not the other (`generate` produces a hald clut, apply can either produce inline or reuse a hald clut and apply it).